lib/_stream_readable.js

```diff
@@ -1070,13 +1072,61 @@ Readable.prototype.wrap = function(stream) {
 };
 
 Readable.prototype[SymbolAsyncIterator] = function() {
+  if (typeof this.read === 'function') {
+    const iter = createAsyncIterator(this);
+    iter.stream = this;
+    return iter;
+  }
+
   if (createReadableStreamAsyncIterator === undefined) {
     createReadableStreamAsyncIterator =
       require('internal/streams/async_iterator');
```

Comment:
We still need `async_iterator` for compat with legacy streams.

Reply:
Why? 

Also, can't we `.wrap` the old stream and then pass them to `createAsyncIterator` instead?

Reply:
Removed